src/app/components/challenge/challenge.component.ts

```diff
@@ -95,4 +101,13 @@ export class ChallengeComponent implements OnInit {
       this.challengeService.starToggle(challengeId);
     }
   }
+
+  /**
+   * Delete click function.
+   */
+  deleteChallenge(challengeId) {
+    if (this.isLoggedIn && this.isHost) {
```

Comment:
It will be better if you write else statement and pop up error - something like ''It looks like you are not logged in or host for this challenge', in case somehow unauthorized user came till here.